File: app/src/main/java/com/tech/dagger2_android/inJava/di/MainViewModelInjector.java
package com.tech.dagger2_android.inJava.di;

import com.tech.dagger2_android.inJava.JavaMainActivity;
import com.tech.dagger2_android.inJava.basic.MainViewModel;

import dagger.Component;

@Component
public interface MainViewModelInjector {
    MainViewModel getMainViewModel();

    void inject(JavaMainActivity javaMainActivity);  //injector factory
}

File: app/src/main/java/com/tech/dagger2_android/inKotlin/di/MainViewModelInjector.kt
package com.tech.dagger2_android.inKotlin.di

import com.tech.dagger2_android.inKotlin.KotlinMainActivity
import com.tech.dagger2_android.inKotlin.basic.MainViewModel
import dagger.Component

@Component
interface MainViewModelInjector {
    //also called component
    fun getMainViewModel(): MainViewModel? //top level factory

    fun injectFields(mainActivity: KotlinMainActivity)
}